components/exercises/chapter-4-oscar/SumOfARange.tsx

```diff
@@ -0,0 +1,211 @@
+import Image from 'next/image';
+import React, { useState } from 'react';
+
+/**
+ * Calculates the sum of a range of numbers.
+ * @param start - The starting number of the range.
+ * @param end - The ending number of the range.
+ * @returns An array containing all the numbers in the range.
+ */
+export const calcRangeSum = (start: number, end: number) => {
+  const range = [];
+  for (let i = start; i <= end; i++) {
+    range.push(i);
+  }
+  return range;
+};
+
+/**
+ * Calculates the sum of a range of numbers based on a given start, end, and step.
+ * @param start - The starting number of the range.
+ * @param end - The ending number of the range.
+ * @param step - The step to increment or decrement the range by.
+ * @returns An array of numbers representing the range.
+ */
+export const calcRangeSumMod = (start: number, end: number, step: number) => {
+  const range = [];
+
+  if (step === 1) {
+    // if step is 1, then we can just use the original function
+    for (let i = start; i <= end; i += step) {
+      range.push(i);
+    }
+  } else if (step < 1) {
```

Comment:
I think you want to check for a negative step, then you should do `step < 0`

Reply:
refactored, no need to add this